File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cs
using System;

public class Solution {
    public int solution(string s) {
        int answer = 0;
        int same = 0;
        int diff = 0;
        char temp = s[0];
        
        for(int i = 0; i < s.Length; i++)
        {
            if(s[i] == temp) same++;
            else diff++;
            
            if(same == diff)
            {
                answer++;
                same = 0;
                diff = 0;
                
                if(i + 1 < s.Length)
                {
                    temp = s[i + 1];
                }
            }
        }
        if(same != 0 || diff != 0) answer++;        
        
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    char c;
    
    int cnt1 = 0, cnt2 = 0;
    
    for(int i = 0; i < s.size(); i++)
    {
        if(cnt1 == 0)
        {
            c = s[i];
        }
        
        if(c == s[i]) cnt1++;
        else cnt2++;
        
        if(cnt1 == cnt2)
        {
            answer++;
            cnt1 = cnt2 = 0;
        }
    }
    if(cnt1 != 0) answer++;
    return answer;
}